helpers/perfMonitor.js

```diff
@@ -0,0 +1,25 @@
+export class PerfMonitor {
+
+	constructor(host) {
+		this.constructedTime = window.performance.now();
+		this.host = host;
+	}
+
+	hostConnected() {
+		this.connectedTime = window.performance.now();
+	}
+
+	hostUpdated() {
+		if (!window.d2lPerfTestInProgress || this.renderedTime !== undefined) return;
```

Comment:
Checking for `d2lPerfTestInProgress` means that in production and in normal use the component isn't firing events all over the place.